classes/class-plugin.php

```diff
@@ -138,6 +145,9 @@ public function __construct() {
 		// Load logger class.
 		$this->log = apply_filters( 'wp_stream_log_handler', new Log( $this ) );
 
+		// Set the IP address for the current request.
+		$this->client_ip_address = wp_stream_filter_input( INPUT_SERVER, 'REMOTE_ADDR', FILTER_VALIDATE_IP );
```

Comment:
Make the main plugin class responsible for knowing properties of the current request. Expose the knowledge via a helper method.